### ISSUE TITLE
Template parts should follow the Synced Pattern interaction model

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#65698]
Now that the concepts around #48458 are more solid, we need to ensure interactions are consistent between inserted template parts, synced patterns, and top level groups through the various edit modes.

- On zoomed-out, controls such as "shuffle" should be present, even if "header" and "footer" are anchored on a template without movers.
- On content-only mode, they should be selectable and expose their child blocks as content.
- If content-only is editable, synced patterns and template parts should always provide the ability to modify content.
- They should all expose hooked blocks as toggleable features: #65676.

**Comment by fabiankaegy** [comment]
> If content-only is editable, synced patterns and template parts should always provide the ability to modify content.

Not sure how I feel about this one. I thought we learned through all the confusion from users that directly allowing editing of synced patterns causes confusion because it's not clear that your change affect the rest of the site? 🤔

**Comment by mtias** [comment]
Yeah, it'd be something we need to clarify on the UI. There's a lot of permutation between partially synced patterns and content-only UI that we need to get right. For fully synced (templates and patterns) it can just be a matter of what is exposed as content—ff there are no named static blocks, for example, perhaps that just doesn't get exposed in content-only. But blocks like site-title, site-logo, navigation, etc, that are already global in attributes would be very useful to have the ability to edit their "content" without having to go into design mode for it.

**Comment by SaxonF** [comment]
The video [here](https://github.com/WordPress/gutenberg/issues/60021) (and attached) touches on this a little bit, specifically at 2.30 mark. As an interim step I think we can still follow the same rules as synced patterns in that editing a template part should require a dedicated "edit" click. This transitions you into a global context. We can do a much better job of highlighting that context in inspector etc.

https://github.com/user-attachments/assets/38a8a789-5682-4c60-93f3-436d61496cd2

The video [here](https://github.com/WordPress/gutenberg/issues/65699#issuecomment-2384913467) is also related and shows highlighting what's editable in inspector.

**Comment by getdave** [comment]
[PR for Experimenting with "Drilldown"](https://github.com/WordPress/gutenberg/pull/64923)

**Comment by sethrubenstein** [comment]
@getdave Is my understanding correct that the new "Drilldown" interaction for content-only mode would allow these blocks to maintain their block context while editing? Currently, this context gets detached when editing patterns in isolation. 

That's really my only concern with changing the template part interaction model is that it could break a few blocks we have that provide context to inner blocks, including template parts.

**Comment by getdave** [comment]
@sethrubenstein I'm afraid I'm not sure at this point. I took a look at that PR again yesterday. It's now fairly out of date and I'm not entirely convinced on the drill down idea. I'm going to close it out until we have more time to explore this. I'm currently focused on WP 6.7 release.

**Comment by talldan** [comment]
I worked a little on #68042, which makes template part interaction more like the pattern block, but I still need to figure out a way to make site title, logo etc editable.
